internal/pkg/status/status_test.go

```diff
@@ -17,19 +17,1001 @@ import (
+	assert.Equal(t, (*status.Condition)(nil), condition)
+
+	r, err = status.GetConditions(y2u(t, podReady))
+	assert.NoError(t, err)
+	ready = status.GetCondition(r, status.ConditionReady)
+	assert.NotEqual(t, nil, ready)
+	assert.Equal(t, "True", ready.Status)
+	assert.Equal(t, "Phase: Running", ready.Reason)
+	condition = status.GetCondition(r, status.ConditionCompleted)
+	assert.Equal(t, (*status.Condition)(nil), condition)
+	condition = status.GetCondition(r, status.ConditionFailed)
+	assert.Equal(t, (*status.Condition)(nil), condition)
+
+	r, err = status.GetConditions(y2u(t, podCompletedOK))
+	assert.NoError(t, err)
+	ready = status.GetCondition(r, status.ConditionReady)
+	assert.NotEqual(t, nil, ready)
+	assert.Equal(t, "True", ready.Status)
+	assert.Equal(t, "Phase: Succeeded, PodCompleted", ready.Reason)
+	condition = status.GetCondition(r, status.ConditionCompleted)
+	assert.NotEqual(t, nil, condition)
+	assert.Equal(t, "True", condition.Status)
+	assert.Equal(t, "Pod Succeeded", condition.Reason)
+	condition = status.GetCondition(r, status.ConditionFailed)
+	assert.Equal(t, (*status.Condition)(nil), condition)
+
+	r, err = status.GetConditions(y2u(t, podCompletedFail))
+	assert.NoError(t, err)
+	ready = status.GetCondition(r, status.ConditionReady)
+	assert.NotEqual(t, nil, ready)
+	assert.Equal(t, "True", ready.Status)
+	assert.Equal(t, "Phase: Failed, PodCompleted", ready.Reason)
+	condition = status.GetCondition(r, status.ConditionCompleted)
+	assert.Equal(t, (*status.Condition)(nil), condition)
+	condition = status.GetCondition(r, status.ConditionFailed)
+	assert.NotEqual(t, nil, condition)
+	assert.Equal(t, "True", condition.Status)
+	assert.Equal(t, "Pod phase: Failed", condition.Reason)
+}
+
+var pvcNoStatus = `
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+   name: test
+`
+var pvcBound = `
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+   name: test
+   namespace: qual
+status:
+   phase: Bound
+`
+
+var pvcUnBound = `
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+   name: test
+   namespace: qual
+status:
+   phase: UnBound
+`
+
+func TestPVCStatus(t *testing.T) {
+	r, err := status.GetConditions(y2u(t, pvcNoStatus))
+	assert.NoError(t, err)
+	ready := status.GetCondition(r, status.ConditionReady)
+	assert.NotEqual(t, nil, ready)
+	assert.Equal(t, "False", ready.Status)
+	assert.Equal(t, "PVC is not Bound. phase: unknown", ready.Reason)
+
+	r, err = status.GetConditions(y2u(t, pvcBound))
+	assert.NoError(t, err)
+	ready = status.GetCondition(r, status.ConditionReady)
+	assert.NotEqual(t, nil, ready)
+	assert.Equal(t, "True", ready.Status)
+	assert.Equal(t, "PVC is Bound", ready.Reason)
+
+	r, err = status.GetConditions(y2u(t, pvcUnBound))
+	assert.NoError(t, err)
+	ready = status.GetCondition(r, status.ConditionReady)
+	assert.NotEqual(t, nil, ready)
+	assert.Equal(t, "False", ready.Status)
+	assert.Equal(t, "PVC is not Bound. phase: UnBound", ready.Reason)
```

Comment:
might want to consider switching this over to an overall `Equals` on the whole condition, or something like `gstruct` from onsi.github.com/gomega